bitcoin/src/blockdata/transaction.rs

```diff
@@ -698,15 +698,20 @@ impl Transaction {
                 .collect(),
             output: self.output.clone(),
         };
-        cloned_tx.txid().into()
+        cloned_tx.compute_txid().into()
     }
 
+    #[deprecated(since = "0.31.0", note = "use compute_txid() instead")]
+    #[doc(alias = "compute_txid")]
+    #[doc(hidden)]
```

Comment:
`doc(hidden)` seems unnecessary and might just confuse people. Deprecation message seems like enough. A docstring explaining to use new name seem more helpful.

Reply:
There's a lint that fails without `doc(hidden)`

Reply:
Yes, `doc(hidden)` is wrong and so is the lint. Which lint is that? Just turn it off and I'll happily complain at the repo of whoever came up with such nonsense..

Reply:
The lint is defined here: https://github.com/rust-bitcoin/rust-bitcoin/blob/master/bitcoin/src/lib.rs#L36

`#![warn(missing_docs)]`

I could change the lint level, but I assumed it was put in place for a reason to ensure a good level of documentation.  Ideally this lint wouldn't complain if there's a `deprecated` attribute..

Reply:
Oh, I didn't notice you didn't write the documentation. You have to add documentation even when it's deprecated. Or maybe especially because people will read it to understand deprecation better, so write a good explanation there.